frame/staking/src/migrations.rs

```diff
@@ -18,6 +18,53 @@
 
 use super::*;
 
+pub mod v8 {
+	use frame_election_provider_support::SortedListProvider;
+	use frame_support::traits::Get;
+
+	use crate::{Config, Nominators, Pallet, StorageVersion, Weight};
+
+	#[cfg(feature = "try-runtime")]
+	pub fn pre_migrate<T: Config>() -> Result<(), &'static str> {
+		ensure!(StorageVersion::<T>::get() == crate::Releases::V7_0_0, "must upgrade linearly");
+		ensure!(T::SortedListProvider::iter().count() == 0, "voter list already exists");
+		crate::log!(info, "👜 staking bags-list migration passes PRE migrate checks ✅",);
+		Ok(())
+	}
+
+	/// Migration to sorted [`SortedListProvider`].
+	/// NOTE: this migration makes the assumption that pallet-bags-list is used as the
+	/// [`SortedListProvider`].
+	pub fn migrate<T: Config>() -> Weight {
+		if StorageVersion::<T>::get() == crate::Releases::V7_0_0 {
+			crate::log!(info, "migrating staking to Releases::V8_0_0");
+
+			let migrated = T::SortedListProvider::regenerate(
+				Nominators::<T>::iter().map(|(id, _)| id),
+				Pallet::<T>::weight_of_fn(),
+			);
```

Comment:
I thought we were doing the migration lazyly ? does this will blow up ?

Reply:
I guess if we are able to read all the nominator in one block we also should be able to write them without costing too much time.
Even if this blow up a bit it shouldn't be critical. (as long as we can have slightly longer block)

Reply:
I think it will be fine to do the migration in one block. The amount of storage read/write is roughly 2x of what we do in a snapshot operation, which should cause a long block but not too much. And we should 100% test that it can be executed in wasm without any memory issues.